### PR TITLE
Add merch_months_in with tests

### PR DESCRIPTION
**Pull request by codevelvet**
Here is a method that returns an array of merch months between start and end dates inclusive.  This is useful for creating time series data from our merch data.

@webdestroya Let me know if you have feedback, thanks!

### CODE REVIEW

File: lib/merch_calendar/date_calculator.rb [hunk 1]
```diff
@@ -98,6 +98,18 @@ def julian_to_merch(julian_month)
       end
     end
 
-
+    def merch_months_in(start_date, end_date)
+      merch_months = []
+      prev_date = start_date - 2
```
Comment (jstrait) [inline lib/merch_calendar/date_calculator.rb]: What is the `- 2` for?

File: spec/merch_calendar/date_calculator_spec.rb [hunk 2]
```diff
@@ -48,5 +48,43 @@
...
+
+      merch_months = subject.merch_months_in(start_date, end_date)
+      expect(merch_months.count).to be(1)
+      expect(merch_months.first.year).to eq start_merch_date.year
+      expect(merch_months.first.month).to eq start_merch_date.month
+      expect(merch_months.first.day).to eq start_merch_date.day
+    end
+
+    it "returns valid merch dates for 2014" do
+      start_date = Date.new(2014, 1, 1)
```
Comment (jstrait) [inline spec/merch_calendar/date_calculator_spec.rb]: Is the expectation that `merch_months_in` will return the next merch months with a start date on or after this start date? I.e. in this example, the first merch month it returns will be the first merch month that starts on or after 1/1/2014?

Comment (codevelvet) [inline spec/merch_calendar/date_calculator_spec.rb]: It's on or after.

Joel Strait wrote:

File: lib/merch_calendar/date_calculator.rb (hunk 1 above)
Comment (codevelvet) [inline lib/merch_calendar/date_calculator.rb]: 2 days. That insures prev_date and date are not equal for the first 
iteration.

Joel Strait wrote:

**Comment by codevelvet** [comment]
@webdestroya  If this is good, I'd like to get it merged so that we can use it in minx.  Thanks.

File: lib/merch_calendar/date_calculator.rb (hunk 1 above)
Comment (davidamcclain) [inline lib/merch_calendar/date_calculator.rb]: Have you thought about replacing the "magic" numbers (2 and 14) with a more descriptive variable?

**Pull request merged**